lib/google_drive/session.rb

```
@@ -86,7 +104,26 @@ def initialize(client_or_access_token, proxy = nil)
         attr_accessor :on_auth_fail
 
         def execute!(*args) #:nodoc:
-          return @fetcher.client.execute!(*args)
+          begin
+            client.execute!(*args)
+          rescue Google::APIClient::ClientError => e
+            if @current_auth_attempts < @max_reauth_attempts
+              response = e.result.response
+              if response.status == 401 && client.authorization.respond_to?(:refresh_token)
+                begin
+                  @current_auth_attempts += 1
+                  resp = client.authorization.fetch_access_token!
+                  post_authorize(resp)
+                  @current_auth_attempts = 0
+                rescue Signet::AuthorizationError
+                  # Ignore since we want the original error
+                end
+                execute!(*args)
+              end
+            else
```

Is there a chance that this could fail silently if `response.status != 401`?